### ISSUE TITLE
Output to a file is choppy

### ISSUE THREAD

**Issue by Kaydax** [Guichaguri/node-chiptune#5]
I have used both lame and wav and still had no hope in finding out why this happens. I pipe have piped the data stream of the file from both local and over the internet using request, and the output file is choppy as all hell. Please help me find out why. Also had another issue as of when using the distributed libopenmpt files, it errors but when updating them it works just fine. This may be playing a role in it, but when piping to the speaker, it outputs without issue.

**Comment by Guichaguri** [comment]
I think it's a problem with the stream implementation.

Because of the way the metadata of these chiptune files work, libopenmpt needs to receive the whole file before it's able to properly convert it.

The duplex stream implementation reads and buffers the whole stream before starting to write data, and that seems pretty broken at the moment.

You can try reading the whole request stream first to a buffer and then creating a chiptune readable stream from it (something similar to the [speaker example](https://github.com/Guichaguri/node-chiptune/blob/master/examples/speaker-example.js)), that should work as expected.

**Comment by Kaydax** [comment]
Yah even when passing in buffered data into the chiptune decoder, it still outputted choppy audio

Current Code:
```javascript
var buffer = fs.readFileSync("{path to mod file}")

var chiptuneDecoder = chiptune(buffer, {
  channels: 2, // 2 channels (stereo)
  sampleRate: 48000 // 48,000 Hz sample rate
});

var lameEncoder = new lame.Encoder({
  // input
  channels: 2,
  bitDepth: 16,
  sampleRate: 48000,

  // output
  bitRate: 128,
  outSampleRate: 48000,
  mode: lame.STEREO
});

var outputStream = fs.createWriteStream('out.mp3');
chiptuneDecoder.pipe(lameEncoder);
lameEncoder.pipe(outputStream);
```

**Comment by Kaydax** [comment]
I also tried exactly what your example did by just using fs.readfile, and it still output the same laggy audio. If you want a preview of what it sounds like I will attach the output
[out.zip](https://github.com/Guichaguri/node-chiptune/files/3519743/out.zip)

**Comment by Kaydax** [comment]
I even tried just outputting the raw PCM data to a file, and it is still choppy, so it isn't anything wrong with the conversion from PCM to mp3 nor just adding the WAVE header with wav.

**Comment by Kaydax** [comment]
Ok so even more testing shows that if I set node-chiptune to process 4 channels, it works without issue and produces clean audio

**Comment by Kaydax** [comment]
Ok so here is the code I have now. I can now output a clean mp3 using lame, but you still need to fix this issue as it's very weird that it does this.
```javascript
//This is the example code using fs.readFile, this can be replaced with request to grab the data

fs.readFile("{input file}", function(err, data) {
  var chiptuneDecoder = chiptune(data, {
    channels: 4, // 2 channels (stereo)
    sampleRate: 48000 // 48,000 Hz sample rate
  });

  var lameEncoder = new lame.Encoder({
    // input
    channels: 2,
    bitDepth: 16,
    sampleRate: 96000,

    // output
    bitRate: 128,
    outSampleRate: 48000,
    mode: lame.STEREO
  });

  var outputStream = fs.createWriteStream('out.mp3');
  chiptuneDecoder.pipe(lameEncoder);
  lameEncoder.pipe(outputStream);
});
```
